File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int top = 0;
        int bottom = 0;
        int prev = -(nums[0] + 1);
        while (top < nums.length) {
            if (nums[top] != prev) {
                nums[bottom] = nums[top];
                bottom++;
            }
            prev = nums[top];
            top++;
        }
        return bottom;
    }
}

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        top = 0
        bottom = 0
        prev = -(nums[0]+1)
        while (top < len(nums)):
            if (nums[top] != prev):
                nums[bottom] = nums[top]
                bottom += 1
            top += 1
            prev = nums[top - 1]
        return bottom
        